client/src/test/java/io/pravega/client/stream/impl/ControllerImplTest.java

```diff
@@ -539,17 +545,49 @@ public void deleteScope(ScopeInfo request, StreamObserver<DeleteScopeStatus> res
             }
         };
 
-        fakeServer = InProcessServerBuilder.forName("fakeserver")
-                .addService(fakeServerImpl)
-                .directExecutor()
+        serverPort = TestUtils.getAvailableListenPort();
+        testGRPCServer = NettyServerBuilder.forPort(serverPort)
+                .addService(testServerImpl)
                 .build()
                 .start();
-        controllerClient = new ControllerImpl(InProcessChannelBuilder.forName("fakeserver").directExecutor());
+        controllerClient = new ControllerImpl(URI.create("tcp://localhost:" + serverPort));
     }
 
     @After
     public void tearDown() {
-        fakeServer.shutdown();
+        testGRPCServer.shutdownNow();
+    }
+
+    @Test
+    public void testKeepAlive() throws IOException, ExecutionException, InterruptedException {
+
+        // Verify that keep-alive timeout less than permissible by the server results in a failure.
+        ControllerImpl controller = new ControllerImpl(NettyChannelBuilder.forAddress("localhost", serverPort)
+                .keepAliveTime(10, TimeUnit.SECONDS).usePlaintext(true));
+        CompletableFuture<Boolean> createStreamStatus = controller.createStream(StreamConfiguration.builder()
+                .streamName("streamdelayed")
+                .scope("scope1")
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build());
+        AssertExtensions.assertThrows("Should throw Exception", createStreamStatus,
+                throwable -> throwable instanceof StatusRuntimeException);
+
+        // Verify that the same RPC with permissible keepalive time succeeds.
+        int serverPort2 = TestUtils.getAvailableListenPort();
+        ServerImpl testServer = NettyServerBuilder.forPort(serverPort2)
+                .addService(testServerImpl)
+                .permitKeepAliveTime(5, TimeUnit.SECONDS)
+                .build()
+                .start();
+        controller = new ControllerImpl(NettyChannelBuilder.forAddress("localhost", serverPort2)
+                .keepAliveTime(10, TimeUnit.SECONDS).usePlaintext(true));
```

Comment:
Also grpc does not accept a value less than 10s, so using the lowest possible value here